fabric/src/main/java/net/william278/husksync/FabricHuskSync.java

```diff
@@ -0,0 +1,325 @@
+
+        try {
+            // Create adventure audience
+            this.audiences = FabricServerAudiences.of(FabricHuskSync.SERVER);
+
+            // Load settings and locales
+            log(Level.INFO, "Loading mod configuration settings & locales...");
+            initialized.set(reload().join());
+            if (initialized.get()) {
+                log(Level.INFO, "Successfully loaded mod configuration settings & locales");
+            } else {
+                throw new HuskSyncInitializationException(
+                        "Failed to load mod configuration settings and/or locales"
+                );
+            }
+
+            // Prepare data adapter
+            if (this.settings.doCompressData()) {
+                this.dataAdapter = new CompressedDataAdapter();
+            } else {
+                this.dataAdapter = new JsonDataAdapter();
+            }
+
+            // Prepare event cannon
+            this.eventCannon = new FabricEventCannon();
+
+            // FIXME: The migration function is not supported.
+
+            // Prepare database connection
+            this.database = new MySqlDatabase(this);
+            log(Level.INFO, "Attempting to establish connection to the database...");
+            initialized.set(this.database.initialize());
+            if (initialized.get()) {
+                log(Level.INFO, "Successfully established a connection to the database");
+            } else {
+                throw new HuskSyncInitializationException("Failed to establish a connection to the database. " +
+                        "Please check the supplied database credentials in the config file");
+            }
+
+            // Prepare redis connection
+            this.redisManager = new RedisManager(this);
+            log(Level.INFO, "Attempting to establish connection to the Redis server...");
+            initialized.set(this.redisManager.initialize());
+            if (initialized.get()) {
+                log(Level.INFO, "Successfully established a connection to the Redis server");
+            } else {
+                throw new HuskSyncInitializationException("Failed to establish a connection to the Redis server. " +
+                        "Please check the supplied Redis credentials in the config file");
+            }
+
+            // Register event handler
+            log(Level.INFO, "Registering events...");
+            this.eventListener = new FabricEventListener(this);
+            log(Level.INFO, "Successfully registered events listener");
+
+            // Register permissions
+            log(Level.INFO, "Registering permissions & commands...");
+
+            // Register commands handler
+            log(Level.INFO, "Successfully registered permissions & commands");
+
+            // FIXME: The plan function is not supported.
+            // FIXME: The bStats metrics function is not supported.
+            // FIXME: The updates function is not supported.
```

Comment:
Totally fine to only support certain features on certain platforms where there's no equivalent.

I think Plan is supported on Fabric? Definitely okay to deprioritize some of this stuff.

Reply:
I overlooked this, Plan is statistically supported by Fabric, I will check their documentation as soon as possible.